posts/hello-world-in-kafka-using-python.md

```diff
@@ -19,30 +19,36 @@ You might have got a hint of it already. This blog is about understanding what i
 ## What is Kafka? Why should one use it?
 In short, Kafka is a distributed streaming and message queueing platform.
 
-> Oh wait! What does that even mean? How different is it from traditional databases? 
+> Oh wait! What does that even mean?
 
+A streaming platform is a system that can perform the following:
+1. Store and process streams of data in real time
+2. Allow applications and/or services to publish and subscribe to these data streams at scale
 
-Simplified explanation of what are streaming and messaging queues and how they are different from traiditional databases.
+> Interesting! How different or similar is it from traditional databases?
 
+Kafka is NOT a database. Nevertheless, it has few similarities and quite an amount of differences with respect to traditional databases.
 
-> That sounds convincing! But why would one need a system like this?
+Like traditional databases, you can store data into Kafka which can be persistent, checksummed and replicated for fault tolerance. This is one reason why Kafka is not just a message queue but a streaming platform. In addition, Kafka has something called as topics which is similar to tables in PostgreSQL or collections in MongoDB. 
 
+Unlike traditional databases, Kafka deals with real time data that can be published or consumed by multiple applications at the same time. Alongside, Kafka allows you to process the data streams but this is totally different when compared to performing CRUD operations or running simple to complex queries on traditional databases. 
 
-## Simplifying the backend architecture.
+> That sounds convincing! But why would one need a system like this?
 
+### 1. Simplify the backend architecture
 
+Look at how a complex architecture can be simplified and streamlined with the help of Kafka
 
-![Without Kafka](https://www.confluent.io/wp-content/uploads/streaming_platform_rev.png)
+![Without Kafka](https://www.confluent.io/wp-content/uploads/data-flow-ugly-1-768x427.png "Without Kafka")
 
+![With Kafka](https://www.confluent.io/wp-content/uploads/data-flow-768x584.png "With Kafka")
 
-![With Kafka](https://www.confluent.io/wp-content/uploads/confluent_platform_diagram.png)
+### 2. Universal pipeline of data
 
-## Adding extensibility for your current backend
-Recover from failures, rebuild existing databases add new systems with less time and effort by using Kafka as the true source od data.
+Kafka is build for scale and thus it ensures that the data is always reliable at any point in time. It also supports strong mechanisms for recovery from failures.
```

Comment:
Really poor explanation for what does it mean by Kafka being used as a universal pipeline of data. Here is what it actually means, where you store data in systems like SQL, Mongo or Cassandra you alter the original data to fit into the modelling of data on these systems, so you don't have the pure source of unaltered data. But do one need the true source of unaltered data? Its because this will help you rebuild an entire database using the true in event of failure, it helps you introduce a new database or a system and use the true source to construct the system for all the historical data too very easily. But once data is convoluted to fit into systems like SQL they are not meant to be pulled to reconstruct or build a new system from scratch.